dashboard/src/components/Header/Header.tsx

```diff
@@ -176,9 +184,8 @@ class Header extends React.Component<IHeaderProps, IHeaderState> {
   private handleNamespaceChange = (ns: string) => {
     const { pathname, push, setNamespace } = this.props;
     const to = pathname.replace(/\/ns\/[^/]*/, `/ns/${ns}`);
-    if (to === pathname) {
-      setNamespace(ns);
-    } else {
+    setNamespace(ns);
```

Comment:
This change just means that `setNamespace` is also called when the path changes, rather than only when it didn't change? Why was it only calling `setNamespace` when it *hadn't* changed in the first place?

Reply:
Yes, the logic was: If we are in a route that doesn't contain the namespace (e.g. /catalog) just call `setNamespace`. In other case, it was relying that a change in the route would trigger the `setNamespace` action by someone else (I assume). But the second was never happening and `setNamespace` was not being called.